x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1221,7 +1245,7 @@ static Expression safeMaybeFold(Expression expression) {
             }
 
             // can it be quickly fast-tracked that the operation can't be reduced?
-            boolean cannotSimplify(BiFunction<DataType, DataType, Boolean> typesCompatible) {
+            boolean isUnsafe(BiFunction<DataType, DataType, Boolean> typesCompatible) {
```

Comment:
If you don't this to be subclass because of the dedicated method, consider making it final.